streaming_form_data/parser.py

```diff
@@ -65,8 +70,12 @@ def data_received(self, data: bytes):
                 message = "internal errors"
             elif ErrorGroup.Delimiting <= result < ErrorGroup.PartHeaders:
                 message = "delimiting multipart stream into parts"
-            elif ErrorGroup.PartHeaders <= result:
+            elif ErrorGroup.PartHeaders <= result < ErrorGroup.UnregisteredPart:
                 message = "parsing specific part headers"
+            elif ErrorGroup.UnregisteredPart <= result:
```

Comment:
```suggestion
            elif ErrorGroup.UnregisteredPart == result:
```

Background: I think I might rework the `ErrorGroup` system to return specific codes for specific errors so we don't have to do `<= result <` style range checks. At the moment those ranges are arbitrary anyway and don't really let the client know what went wrong. So this suggestion would make that work somewhat easier. :)